packages/tools/client-debugger/client-debugger-chrome-extension/src/react-components/AudienceView.tsx

```diff
@@ -25,12 +34,58 @@ export interface AudienceViewProps extends HasContainerId {
 export function AudienceView(props: AudienceViewProps): React.ReactElement {
 	const { containerId } = props;
 
-	// TODO: Post message requesting Audience summary
-	// TODO: Listen for Audience data updates
+	const messageRelay = React.useContext(MessageRelayContext);
+
+	if (messageRelay === undefined) {
+		throw new Error(
+			"MessageRelayContext was not defined. Parent component is responsible for ensuring this has been constructed.",
+		);
+	}
+
+	const [allAudienceMembers, setAllAudienceMembers] = React.useState<
+		Map<string, IClient> | undefined
+	>();
+	const [audienceHistory, setAudienceHistory] = React.useState<readonly unknown[]>();
 
 	React.useEffect(() => {
-		// TODO
-	}, [containerId]);
+		/**
+		 * Handlers for inbound messages related to Audience
+		 */
+
+		const inboundMessageHandlers: InboundHandlers = {
+			["AUDIENCE_EVENT"]: (untypedMessage) => {
+				const message: AudienceEventMessage = untypedMessage as AudienceEventMessage;
+				console.log("Passed 3");
+				console.log(message);
+
+				// setAllAudienceMembers([message.data.audienceState]);
+				setAudienceHistory([message.data.audienceHistory]);
```

Comment:
`message.data.audienceHistory` is already an array, right? Isn't this putting an array-of-arrays into the local state variable?

Reply:
Fixed to `setAudienceHistory(message.data.audienceHistory);`